docs/boot-options.rst

```diff
@@ -514,6 +514,36 @@ Specify screen size for the installer. Use format nxm, where n is the
 number of horizontal pixels, m the number of vertical pixels. The lowest
 supported resolution is 800x600.
 
+.. inst.rdp:
+
+inst.rdp
+^^^^^^^^
+
+Enable Remote Desktop Protocol-controlled installation. You will need to connect to
+the machine using an RDP client application. An RDP install implies that the installed
+system will boot up in in multiuser.target instead of to the graphical login screen.
+
+Multiple RDP clients can connect.
+
+When using ``inst.rdp``, you also need to set RDP username and password using the
+``inst.rdp.username`` and ``inst.rdp.password`` boot options.
+
+.. inst.rdp.username:
+
+inst.rdp.username
+^^^^^^^^^^^^^^^^^
+
+Set username for the RDP session. To enable RDP access, also use the
+``inst.rdp`` and ``inst.rdp.password`` boot options.
+
+.. inst.rdp.password:
+
+inst.rdp.password
+^^^^^^^^^^^^^^^^^
+
+Set password for the RDP session. To enable RDP access, also use the
+``inst.rdp`` and ``inst.rdp.username`` boot options.
```

Comment:
I wonder if we need user to specify all of these options. I mean what would be the use-case for having `inst.rdp` without password and username -> invalid. On the same note what would be the use case for user to have `inst.rdp.username` without `inst.rdp`.

Reply:
So this is what happens if you use just `inst.rdp`:
![inst rdp_only](https://github.com/rhinstaller/anaconda/assets/829558/f34e7d41-7bed-4005-8e93-393000e6b61f)
If you use just `inst.username`, it boots to GUI without warning - I think that's wrong & I'll fix it.
If you use `inst.rdp` and `inst.username`, you get:
![inst rdp_inst username](https://github.com/rhinstaller/anaconda/assets/829558/101a3019-2963-4280-89ec-d8443e5374ac)
If you use just `inst.password`, it boots to GUI without warning - I think that's wrong & should be fixed as well.
If you use `inst.rdp` and `inst.password`, you get:
![inst rdp_inst password](https://github.com/rhinstaller/anaconda/assets/829558/36e8fc75-c85c-497c-8289-339a2412d9bf)

So indeed, this should be fixed up and simplified. I suggest:

- we drop the lone `inst.rdp` from docs & make sure there is an error when it is used (as we can't avoid people just typing that into the boot options)
- show an error when only `inst.password` or `inst.username` is used

@jkonecny12 @jstodola does it make sense like this ? :)

Reply:
The use case for just `inst.rdp` is to be able to specify the username and password interactively. You might not want to specify the password on the kernel command line. IMO, it makes sense to keep the boot option.

Looking at the current `inst.vnc` and `inst.vncpassword` behavior,  having just `inst.vncpassword` itself on the kernel command line doesn't start VNC mode. So, my suggestion is to keep this logic and require `inst.rdp` for remote graphical installations.

Perhaps if `inst.rdp` is specified on the kernel cmdline and one of `inst.rdp.username` or `inst.rdp.password` is missing, anaconda should ask for the missing value interactively, while respecting the other value specified on the kernel cmdline.

Reply:
I like that idea! I've adjusted the interactive code accordingly:

- if `inst.rdp` is used, go directly to the interactive entry of RDP credentials
- if username or password is set via boot option, ask just for the other bit
- only using `inst.rdp.username` or `inst.rdp.password` without also `inst.rdp` should not do anything, same as with VNC so far
- also some more minor cleanups and UX improvements

![better_interactive_auth](https://github.com/rhinstaller/anaconda/assets/829558/0cbbd59f-51ea-4a8a-9950-2a431b91e556)

You might notice some other stuff writing over the console - definitely something to investigate but IMHO not blocking.

I've created also a testing image - just boot the first boot options & you should go straight to the interactive entry:
https://fedorapeople.org/groups/anaconda/wayland/gnome_remote_desktop/better_rdp_interactive_auth_mk1.iso